src/include/serial_inline.h

```diff
@@ -128,7 +128,7 @@ __col_append_serial_func(WT_SESSION_IMPL *session, WT_INSERT_HEAD *ins_head, WT_
      */
     *recnop = recno;
     if (recno > btree->last_recno)
-        btree->last_recno = recno;
+        __wt_atomic_cas64(&btree->last_recno, btree->last_recno, recno);
```

Comment:
I believe this CAS needs to be in a loop, what we want is that the largest recno wins. So if we fail at swapping in our recno we need to re-read the last_recno and if we're still bigger try agan.

Reply:
Yup good point.